code/modules/projectiles/guns/projectile/constructable/swords.dm

```diff
@@ -45,33 +45,48 @@
 			new /obj/item/weapon/sword/executioner(get_turf(src.loc))
 		qdel(src)
 		qdel(W)
-	if(istype(W, /obj/item/weapon/reagent_containers/hypospray))
+	if(istype(W, /obj/item/weapon/reagent_containers/hypospray) && W.type != /obj/item/weapon/reagent_containers/hypospray/autoinjector)
```

Comment:
You should just check type explicitly in this case. W.type == hypospray type. This avoids legacy issues if someone adds another type of injector.